### PR TITLE
General improvements within readability, syntax, and typos.

### PR DESCRIPTION
**Pull request by BenjaminKjopKofoed**

### CODE REVIEW

File: CadRevealComposer/Operations/SectorSplitting/TooFewInstancesHandler.cs [hunk 1]
```diff
@@ -58,15 +57,12 @@ private TriangleMesh ConvertInstanceToMesh(InstancedMesh instanceMesh)
...
     {
-        int numberOfInstancesThreshold = 100; // Always keep when the number of instances is exceeding the threshold
-        int numberOfTrianglesThreshold = 10000; // Alwyas keep when the number of triangles is exceeding the threshold
+        const int numberOfInstancesThreshold = 100; // Always keep when the number of instances is exceeding the threshold
+        const int numberOfTrianglesThreshold = 10000; // Always keep when the number of triangles is exceeding the threshold
 
-        float a = 100; // Steepness
-
-        if (a < 0)
-            throw new ArgumentException($"The value of A needs to be larger than zero. It was: {a}");
```
Comment (Strepto) [inline CadRevealComposer/Operations/SectorSplitting/TooFewInstancesHandler.cs:69]: Consider moving the exception comment to the comment on steepness.

File: CadRevealComposer/Operations/Tessellating/TessellationUtils.cs [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-using CadRevealComposer.Tessellation;
-using CadRevealComposer.Utils;
+using Primitives;
 using System;
 using System.Collections.Generic;
 using System.Drawing;
 using System.Linq;
 using System.Numerics;
+using Tessellation;
+using Utils;
```
Comment (Strepto) [inline CadRevealComposer/Operations/Tessellating/TessellationUtils.cs:10]: Pretty sure the import sorting order is wrong? 🤔

Comment (Strepto) [reply CadRevealComposer/Operations/Tessellating/TessellationUtils.cs:10]: Consider updating csharpier to the latest version and running that on all files.

**Review by Strepto — APPROVED** [review]

File: CadRevealComposer.Exe/README.md [hunk 3]
```diff
@@ -60,7 +60,7 @@ As of now Cad Reveal Composer recursively splits space into eight subvoxels usin
 
 #### Geometry: tesselation
```
Comment (vegasten) [inline CadRevealComposer.Exe/README.md:61]: There's a missing L here too?

File: CadRevealComposer.Tests/Operations/Tessellating/CircleTessellatorTests.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ public void TessellateCircle_ReturnsCorrectNumberOfVerticesAndIndices()
         var dummyBoundingBox = new BoundingBox(Vector3.Zero, Vector3.Zero);
         var circle = new Circle(Matrix4x4.Identity, Vector3.UnitY, 1, Color.Red, dummyBoundingBox);
 
-        var tessellatedCircle = CircleTessellator.Tessellate(circle)!;
```
Comment (vegasten) [inline CadRevealComposer.Tests/Operations/Tessellating/CircleTessellatorTests.cs:17]: Is this necessary?

File: CadRevealComposer/ModelMetadata.cs [hunk 5]
```diff
@@ -1,38 +1,33 @@
...
-using System.Text.Json.Serialization;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Text.Json;
 
 public class ModelMetadata
 {
-    Dictionary<string, string> metadata;
+    readonly Dictionary<string, string> _metadata;
```
Comment (vegasten) [inline CadRevealComposer/ModelMetadata.cs:8]: Maybe set it to private explicitly?

File: CadRevealComposer/ModelMetadata.cs [hunk 6]
```diff
@@ -1,38 +1,33 @@
...
+    public ModelMetadata(Dictionary<string, string> metadata)
     {
-        metadata = _metadata;
+        this._metadata = metadata;
     }
 
     public int Count()
     {
-        return this.metadata.Count;
+        return this._metadata.Count;
```
Comment (vegasten) [inline CadRevealComposer/ModelMetadata.cs:17]: "this" should not be necessary here

File: CadRevealComposer/ModelMetadata.cs [hunk 7]
```diff
@@ -1,38 +1,33 @@
...
     }
 
     public void Add(ModelMetadata modelMetadata)
     {
-        foreach (var kvp in modelMetadata.metadata)
+        foreach (var kvp in modelMetadata._metadata)
```
Comment (vegasten) [inline CadRevealComposer/ModelMetadata.cs:22]: Wait, I just filled a hole in my c# knowledge. The private access modifier is class based not object based 🤯

File: CadRevealComposer/ModelMetadata.cs [hunk 8]
```diff
@@ -1,38 +1,33 @@
...
         {
             // This will throw if the key already exists!:)
-            this.metadata.Add(kvp.Key, kvp.Value);
+            this._metadata.Add(kvp.Key, kvp.Value);
```
Comment (vegasten) [inline CadRevealComposer/ModelMetadata.cs:25]: "this" should not be necessary here either

File: CadRevealRvmProvider.Tests/Converters/CapVisibilityTests.cs [hunk 9]
```diff
@@ -775,7 +774,7 @@ public void CalculateCapVisibility_TestSnoutCapMatch_Cylinder()
     public void CalculateCapVisibility_TestSnoutCapMatch_RotatedCylinders()
     {
         var transform1 = Matrix4x4.CreateTranslation(0.0f, 0.0f, 0.0f);
-        var rotate = Matrix4x4.CreateRotationX(0.1f);
+        // var rotate = Matrix4x4.CreateRotationX(0.1f);
```
Comment (vegasten) [inline CadRevealRvmProvider.Tests/Converters/CapVisibilityTests.cs:777]: Just remove, I think

File: CadRevealRvmProvider.Tests/Converters/RvmCylinderConverterTests.cs [hunk 10]
```diff
@@ -9,7 +9,7 @@
 [TestFixture]
 public class RvmCylinderConverterTests
 {
-    const int _treeIndex = 1337;
+    const int TreeIndex = 1337;
```
Comment (vegasten) [inline CadRevealRvmProvider.Tests/Converters/RvmCylinderConverterTests.cs:12]: Maybe set it to private explicitly?

File: CadRevealRvmProvider.Tests/Converters/RvmBoxConverterTests.cs [hunk 11]
```diff
@@ -9,7 +9,7 @@
 [TestFixture]
 public class RvmBoxConverterTests
 {
-    const int _treeIndex = 1337;
+    const int TreeIndex = 1337;
```
Comment (vegasten) [inline CadRevealRvmProvider.Tests/Converters/RvmBoxConverterTests.cs:12]: Maybe set it to private explicitly?

**Review by vegasten — APPROVED** [review]

File: CadRevealRvmProvider.Tests/Converters/RvmBoxConverterTests.cs (hunk 11 above)
Comment (vegasten) [reply CadRevealRvmProvider.Tests/Converters/RvmBoxConverterTests.cs:12]: If you decide it is worth it, there are a few other similar cases in the converter tests as well

**Review by Strepto — CHANGES_REQUESTED** [review]
There are a lot of nice changes here, with some changes I would like removed.

But this branch has waaaaay too many changes now. I'll try to review them when I have the time, but you should have stopped for a review after 10 files not 100 ... 😅 

A lot of the naming renames are OK, but some places it reduces readability of the hard math some places. We should disable and enable auto-renames for those files.

The eager use of IEnumerable everywhere instead of the Array type is a possible performance-waste, and is a rider quirk we need to ignore in many places.

Some unused stuff have been removed, and needs more review for if we should remove.

I think the best way to merge this is to split the fixes into categories, and apply them in separate PRs.

File: CadRevealComposer/ModelMetadata.cs (hunk 7 above)
Comment (BenjaminKjopKofoed) [reply CadRevealComposer/ModelMetadata.cs:22]: YEEEEP

File: CadRevealComposer.Tests/Operations/Tessellating/CircleTessellatorTests.cs (hunk 4 above)
Comment (BenjaminKjopKofoed) [reply CadRevealComposer.Tests/Operations/Tessellating/CircleTessellatorTests.cs:17]: No

**Comment by Strepto** [comment]
Closing this as may changes has been done in other branches now.

**Pull request closed without merging**